t/unit/worker/test_strategy.py

```diff
@@ -234,6 +234,35 @@ def test_signal_task_received(self):
                                              request=ANY,
                                              signal=signals.task_received)
 
+    def test_log_task_received_meta(self, caplog):
+        class Formatter:
+            def __init__(self, base_formatter):
+                self.base_formatter = base_formatter
+
+            def format(self, record):
+                record.foo = "bar"
+                return self.base_formatter.format(record)
+
+        old_formatter = caplog.handler.formatter
+
+        def callback(*args, **kwargs):
+            caplog.handler.setFormatter(Formatter(old_formatter))
+
+        caplog.set_level(logging.INFO, logger="celery.worker.strategy")
+
+        with self._context(
+            self.add.s(2, 2)
+        ) as C:
+            signals.task_received.connect(callback)
+            C()
+        for record in caplog.records:
+            if record.msg == LOG_RECEIVED:
+                assert record.foo == "bar"
+                caplog.handler.setFormatter(old_formatter)
+                break
+        else:
```

Comment:
I am not sure if this else is properly aligned within a if else statement block?

Reply:
This is not if/else construction. It's for/else. That's mean - if not break inside "for" block - go to the else block.

Examples can be found here https://book.pythontips.com/en/latest/for_-_else.html#else-clause

Reply:
that's nice! btw can you check the PYPY failure please?